### ISSUE TITLE
`RedundantFunctionCall` reported when calling `strtolower` on stringable object

### ISSUE THREAD

**Issue by derrabus** [vimeo/psalm#7611]
Given the following code:

```PHP
class Foo
{
    public function __toString(): string
    {
        return 'FOO';
    }
}

function lowerFoo(Foo $foo): string
{
    return strtolower($foo);
}
```

https://psalm.dev/r/76b99995b2

Psalm started to report a `RedundantFunctionCall` in addition to the previously reported (and perfectly valid) `ImplicitToStringCast` for the `strtolower()` call. From my point of view, the new error is wrong and should not be reported. The call is not redundant at all and removing it would change the behavior of the code significantly.

The correct course of action is to explicitly cast the argument to `string` which makes both errors go away. However, the `ImplicitToStringCast` error suggests that solution.

**Comment by AndrolGenhald** [comment]
Duplicate of #7536.

**Comment by derrabus** [comment]
Indeed, thanks!
